### DISCUSSION TITLE
Multicast not working, how to configure?

### DISCUSSION THREAD

**Discussion by carlosag012** [bluenviron/mediamtx#609] [Questions]
Ok, I had to edit the whole issue because I was not right. Starting with Docker, I was able to modify the config file. My only change `multicastIPRange: 231.192.16.30/32`
Then started the server with:
`docker run --rm -it -e RTSP_PROTOCOLS=tcp -p 8554:8554 -p 1935:1935 -p 8002:8002 -v $PWD/rtsp-simp
le-server.yml:/rtsp-simple-server.yml aler9/rtsp-simple-server`
and tried both these streams: 
udp://@231.192.16.30 and udp://@231.192.16.30/compressed
from a computer connected to the same LAN, but no multicast. Unicast at the RTSP stream works fine. 
Also, tried the same thing with the Windows distribution and no luck either. 
Am I missing something? Could you please point me in the right direction?
My goal is to have the server in a docker container multicasting the rtsp stream to the given address. 

Thank you

**Comment by aler9** [comment]
Hello, in order to multicast a stream, you have to do this:

1. start the server with the --network=host flag:

   ```
   docker run --rm -it --network=host aler9/rtsp-simple-server
   ```

2. publish to the server the stream you want to multicast:

   ```
   ffmpeg -re -stream_loop -1 -i file.ts -c copy -f rtsp rtsp://localhost:8554/mystream
   ```

3. read that stream with ffmpeg and the RTSP-multicast protocol:

   ```
   ffmpeg -rtsp_transport udp_multicast -i rtsp://localhost:8554/mystream -c copy -f mpegts -y /dev/null
   ```

The client will keep a direct connection to the server only to send the handshake and send keepalives, while the server will stream the video to one of the multicast IP provided in the configuration file. In case of multiple clients, the server will stream the video once, and tell all the clients to read the stream from a single multicast IP and port, saving bandwidth.

You can't simply read from udp://multicast_ip:8000 since:

* this works for a single track only, while RTSP supports infinite tracks
* you don't know the codec of the track and its parameters, so you can't decode it. A RTSP server sends the codec and its parameter to the client in the handshake

**Reply by carlosag012** [reply]
Thanks for the answer. I've tried this and see the multicast leaving the server, but can't get the stream in vlc. When I try rtsp://ip:8554/mystream I see the connection going to the server's ip and port 8000, but this is not multicast.
 When I try to enable port 8002 for multicast, the server crashes.
Let me know if there is anything else that can be done? and thanks again for the support.

**Reply by aler9** [reply]
The readme has been improved with a Multicast section, please read it to get VLC working with UDP multicast:

https://github.com/aler9/rtsp-simple-server#multicast

[a comment by an automated account is not shown]
